websites/D/Duolingo/presence.ts

```diff
@@ -1,92 +1,253 @@
+  if (!course) return;
+
+  const languageId = course.learningLanguage;
+  if (FLAG_ICONS.includes(languageId)) language.imageKey = `flag_${languageId}`;
+  else if (languageId) language.imageKey = "flag_unknown";
+  else language.imageKey = null;
+  language.name = course.title ?? null;
+}
+updateLanguage();
+setInterval(updateLanguage, 1000);
+
+function checkBasicPages(path: string[]): boolean {
+  if (path.length === 0) {
     if (
-      path1.length > 1 &&
-      path1.split("/")[2] !== null &&
-      path1.split("/")[2].length === 2
+      new URL(document.location.href).searchParams.get("isLoggingIn") === "true"
     ) {
-      let language: string;
-      for (const value of lang.keys()) {
-        if (path1.split("/")[2] === value) {
-          language = lang.get(value);
-          break;
-        }
-      }
-      presenceData.details = "Taking a lesson";
-      presenceData.state = `Language: ${language}`;
-      presenceData.largeImageKey = "logo";
-      presence.setActivity(presenceData);
+      presenceData.details = "Logging in";
+      return true;
+    }
+    presenceData.details = "Viewing the home page";
+    return true;
+  }
+  if (INFO_PAGES.includes(path[0])) {
+    presenceData.details = `Viewing the ${path[0]} page`;
+    return true;
+  }
+  if (path[0] === "courses") {
+    presenceData.details = "Viewing available courses";
+    return true;
+  }
+  if (path[0] === "abc") {
+    presenceData.details = "Looking into Duolingo ABC";
+    return true;
+  }
+  if (path[0] === "plus") {
+    presenceData.details = "Looking into Duolingo Plus";
+    return true;
+  }
+  if (path[0] === "dictionary") {
+    presenceData.details = "Looking up a word";
+    if (path.length >= 3) presenceData.state = `${path[1]}: ${path[2]}`;
+    return true;
+  }
+  if (path[0] === "profile") {
+    if (path.length < 2) return true;
+    presenceData.details = "Viewing a profile";
+    [, presenceData.state] = path;
+    if (path.length >= 3) {
+      presenceData.details += " section";
+      presenceData.state += `'s ${path[2]}`;
+    }
+    return true;
+  }
+  if (path[0] === "friend-updates") {
+    presenceData.details = "Viewing friend updates";
+    return true;
+  }
+  if (path[0] === "user-search") {
+    presenceData.details = "Searching for a user";
+    return true;
+  }
+  if (path[0] === "settings") {
+    presenceData.details = "Adjusting settings";
+    if (path.length >= 2) {
+      let [, page] = path;
+      page = page.charAt(0).toUpperCase() + page.slice(1);
+      presenceData.state = `Section: ${page}`;
+    }
+    return true;
+  }
+  if (document.title.startsWith("Error")) {
+    presenceData.details = "Watching Duo cry :(";
+    presenceData.state = document.title;
+    return true;
+  }
+  if (API_ENDPOINTS.includes(path[0])) {
+    presenceData.details = "Viewing an API response";
+    presenceData.state = `Endpoint: /${path[0]}`;
+    return true;
+  }
+
+  return false;
```

Comment:
Use else if

Reply:
I can definitely do that if you want, but it makes no difference since I `return` in all of the `if`s anyway.

Reply:
style